create_derived_table_domains_source/source.py

```diff
@@ -0,0 +1,78 @@
+from typing import Iterable
+
+import json
+import datahub.emitter.mce_builder as builder
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.api.decorators import (
+    SupportStatus,
+    config_class,
+    platform_name,
+    support_status,
+)
+from datahub.ingestion.api.source import (
+    SourceReport,
+    Source,
+)
+
+from datahub.metadata.schema_classes import DomainPropertiesClass, ChangeTypeClass
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.ingestion.api.workunit import MetadataWorkUnit
+
+
+from .config import CreateDerivedTableDomainsConfig
+
+
+@config_class(CreateDerivedTableDomainsConfig)
+class CreateDerivedTableDomainsSource(Source):
+    source_config: CreateDerivedTableDomainsConfig
+    report: SourceReport = SourceReport()
+
+    def __init__(self, config: CreateDerivedTableDomainsConfig, ctx: PipelineContext):
+        super().__init__(ctx)
+        self.source_config = config
+
+    @classmethod
+    def create(cls, config_dict, ctx):
+        config = CreateDerivedTableDomainsConfig.parse_obj(config_dict)
+        return cls(config, ctx)
+
+    def get_workunits(self) -> Iterable[MetadataWorkUnit]:
+        if self.source_config.manifest_s3_uri is None:
+            with open(self.source_config.manifest_local_path, "r") as f:
+                manifest = json.load(f)
+        else:
+            # TODO: implement get from s3
```

Comment:
Can this raise NotImplementedError so if the config is set it doesn't fail silently?